src/lib.rs

```diff
@@ -8,7 +8,7 @@ use rand::{seq::SliceRandom, thread_rng};
 /// single UTXO, or a group of UTXOs that should be spent together.
 /// The library user is responsible for crafting this structure correctly. Incorrect representation of this
 /// structure will cause incorrect selection result.
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
```

Comment:
Multiple UTXOs can have same OutGroup, so deriving Equality doesn't make sense here.

Reply:
Agreed. However, I need the PartialEq trait as I use it to compare the sorted and original OutputGroup vectors to extract the indices from the original OutputGroup vector.

Reply:
But that's not semantically right, and your algorithm is running in O(n^2) while it should be optimally run in O(n logn) due to sorting.
You have to collect the elements with their indices using enumeration before you sort then later sort them according to the creation_sequence.
You an check how I did that in SRD.

Reply:
Understood. Made relevant changes and pushed a commit